jerry-ext/include/jerryscript-ext/section.h

```diff
@@ -0,0 +1,81 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRY_EXT_SECTION_H
```

Comment:
Note 1: Do we want to expose the contents of this header to library users as an extension on its own, or we just want to (potentially) share it between public headers of extensions? To me, it does not look like a jerryscript-related extension but rather a very low-level technicality needed by some extensions. If that's the case, we have an established precedent of naming such a file as xxx.impl.h. (Note that in that case public markdown doc is not needed, even superfluous I'd say.)

Note 2: header macro guard should be JERRYX_xxx_H (i.e., JERRYX_SECTION_H or JERRYX_SECTION_IMPL_H)